ocs_ci/ocs/utils.py

```diff
@@ -1284,3 +1284,48 @@ def thread_init_class(class_init_operations, shutdown):
         if shutdown == 0:
             executor.shutdown(wait=True)
             return
+
+
+def collect_pod_container_rpm_package(dir_name):
+    """
+    Collect information about rpm packages from all containers
+
+    Args:
+        dir_name(str): directory to store container rpm package info
+
+    """
+    # Import pod here to avoid circular dependency issue
+    from ocs_ci.ocs.resources import pod
+
+    timestamp = time.time()
+    cluster_namespace = ocsci_config.ENV_DATA["cluster_namespace"]
+
+    log_dir_path = os.path.join(
+        os.path.expanduser(ocsci_config.RUN["log_dir"]),
+        f"{dir_name}_{ocsci_config.RUN['run_id']}",
+    )
+    package_log_dir_path = os.path.join(
+        log_dir_path, "rpm_package", f"rpm_list_{timestamp}"
+    )
+    create_directory_path(package_log_dir_path)
+    log.info(f"Directory path for rpm logs is{package_log_dir_path}")
+    pods = pod.get_all_pods(namespace=cluster_namespace)
+    for pod_name in pods:
+        pod_object = pod_name.get()
+        pod_containers = pod_object.get("spec").get("containers")
+        for container in pod_containers:
+            container_name = container["name"]
+            ocp_pod_obj = OCP(kind=constants.POD, namespace=cluster_namespace)
+            pod_status = ocp_pod_obj.get_resource_status(pod_name.name)
+            if pod_status == constants.STATUS_RUNNING:
```

Comment:
we need to check these before the for loop, there is no need to check for every container